src/pages/home/sidebar/SidebarLinks.js

```diff
@@ -104,58 +123,63 @@ class SidebarLinks extends React.Component {
+        // Always update if LHN is empty.
+        if (orderedReports.length === 0) {
+            return true;
+        }
+
+        const didActiveReportChange = currentlyViewedReportID !== nextProps.currentlyViewedReportID;
 
         // Always re-order the list whenever the active report is changed
         if (didActiveReportChange) {
             return true;
         }
 
-        const previousDraftComments = this.props.draftComments;
-        const nextDraftComments = nextProps.draftComments;
-
-        const previousDraftReports = Object.keys(previousDraftComments);
-        const nextDraftReports = Object.keys(nextDraftComments);
-
-        const reportsWithNewDraftComments = nextDraftReports.filter((report) => {
-            const isNewDraftComment = !previousDraftReports.includes(report);
-            const wasNonEmptyDraftComment = previousDraftComments[report] === '';
-            const hasDraftCommentChanged = previousDraftComments[report] !== nextDraftComments[report];
-
-            return isNewDraftComment || (hasDraftCommentChanged && wasNonEmptyDraftComment);
-        });
-        const reportsWithRemovedDraftComments = previousDraftReports.filter((report) => {
-            const isRemovedDraftComment = !nextDraftReports.includes(report);
-            const isEmptyDraftComment = nextDraftComments[report] === '';
-            const hasDraftCommentChanged = previousDraftComments[report] !== nextDraftComments[report];
-
-            return isRemovedDraftComment || (hasDraftCommentChanged && isEmptyDraftComment);
-        });
-        const reportsWithEditedDraftComments = nextDraftReports.filter((report) => {
-            const didDraftCommentExistPreviously = previousDraftReports.includes(report);
-            const hasDraftCommentChanged = previousDraftComments[report] !== nextDraftComments[report];
-            const isNotANewDraftComment = !reportsWithNewDraftComments.includes(report);
-            const isNotARemovedDraftComment = !reportsWithRemovedDraftComments.includes(report);
-
-            return didDraftCommentExistPreviously && hasDraftCommentChanged && isNotANewDraftComment && isNotARemovedDraftComment;
-        });
-
-        const allReportsWithDraftCommentChanges = [
-            ...reportsWithNewDraftComments,
-            ...reportsWithRemovedDraftComments,
-            ...reportsWithEditedDraftComments,
-        ];
-
-        const activeReportID = this.props.currentlyViewedReportID;
-        const reportKey = `${ONYXKEYS.COLLECTION.REPORT_DRAFT_COMMENT}${activeReportID}`;
-
-        // Do not re-order reports if draft comment changes are only in the current report.
-        if (allReportsWithDraftCommentChanges.length === 1 && allReportsWithDraftCommentChanges.includes(reportKey)) {
-            return false;
+        // If any reports have new unread messages, re-order the list
+        const nextUnreadReports = SidebarLinks.getUnreadReports(nextProps.reports);
```

Comment:
Think we should check for null here first to avoid any errors if there would be reports missing (it is probably very unlikely, but better to have ti here, than inside of the function.
```suggestion
        const nextUnreadReports = SidebarLinks.getUnreadReports(nextProps.reports || {});
```